src/screens/CreatePost/index.js

```diff
@@ -506,14 +509,23 @@ const CreatePost = () => {
         });
       }
 
-      await createPost(data);
+      const post = await createPost(data);
       if (params.onRefresh && typeof params.onRefresh === 'function') {
         params.onRefresh();
       }
-      showMessage({
-        message: StringConstant.createPostDone,
-        type: 'success'
-      });
+      if (post.code === 200) {
+        showMessage({
+          message: StringConstant.createPostDone,
+          type: 'success'
+        });
+        setLoadingPost(false);
+      } else {
+        showMessage({
+          message: StringConstant.createPostFailedGeneralError,
+          type: 'danger'
+        });
+        setLoadingPost(false);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this block, the `createPost` function is awaited and its result is checked. If the post creation was successful (code 200), a success message is shown and `loadingPost` is set to false. If not, an error message is shown and `loadingPost` is also set to false. This is a good practice as it ensures the loading state is correctly managed regardless of the outcome of the post creation. However, there should be a try-catch block around `createPost(data)` to handle any exceptions that might occur during the execution of this function and to ensure `setLoadingPost(false)` is called in case of an exception.
```diff
+      const post = await createPost(data);
+      if (params.onRefresh && typeof params.onRefresh === 'function') {
+        params.onRefresh();
+      }
+      if (post.code === 200) {
+        showMessage({
+          message: StringConstant.createPostDone,
+          type: 'success'
+        });
+        setLoadingPost(false);
+      } else {
+        showMessage({
+          message: StringConstant.createPostFailedGeneralError,
+          type: 'danger'
+        });
+        setLoadingPost(false);
+      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->